RavenDB.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -33,12 +33,14 @@ public static IServiceCollection AddRavenDbDocStore(this IServiceCollection serv
             this IServiceCollection services,
             Action<RavenOptions> options)
         {
-            services.ConfigureOptions<RavenOptionsSetup>();
             services.Configure(options);
+            services.ConfigureOptions<RavenOptionsSetup>();
             services.AddSingleton(sp =>
             {
                 var setup = sp.GetRequiredService<IOptions<RavenOptions>>().Value;
-                return setup.GetDocumentStore(setup.BeforeInitializeDocStore);
+                var docStore = setup.GetDocumentStore(setup.BeforeInitializeDocStore);
+                setup.AfterInitializeDocStore(docStore);
```

Comment:
Woops - this caused a null reference exception in projects where no AfterInitializeDocStore was setup. I've updated the code to check for this first.